Use https Url for PoliciesTemplateUri in API Stack - develop

jdeamicis opened the pull request:
### Description of changes
* Set default URL for the policies stack template as a https URL instead of a s3 URL (apparently not supported by the CFN CreateStack API).

### Tests
* Successfully deployed the API stack by overriding the `PoliciesTemplateUri` using the https URL of the same S3 object currently specified via a s3 URL.

### References
* https://github.com/aws/aws-parallelcluster/pull/5250

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [ ] ~Make sure **to have added unit tests or integration tests** to cover the new/modified code.~
- [ ] ~Check if documentation is impacted by this change.~

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

jdeamicis committed:
Use https Url for PoliciesTemplateUri in API Stack

The CloudFormation CreateStack API apparently does not support s3 URLs
for the TemplateURL parameter. Therefore a https URL must be used
in our API CFN template to avoid failures in the policies sub-stack
creation.

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>

enrico-usai commented on api/infrastructure/parallelcluster-api.yaml:
```diff
@@ -166,7 +166,7 @@ Resources:
         - !Ref PoliciesTemplateUri
         - !Sub
-          - "s3://${AWS::Region}-aws-parallelcluster/parallelcluster/${Version}/templates/policies/policies.yaml"
+          - "https://${AWS::Region}-aws-parallelcluster.s3.amazonaws.com/parallelcluster/${Version}/templates/policies/policies.yaml"
```
This doesn't work in partitions other than commercial, check `get_templates_bucket_path`

jdeamicis committed:
Generalize https Url for PoliciesTemplateUri in API Stack

Use CloudFormation pseudo parameters to compose the https URL for the
policies stack template.
See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>
